crates/gossip/proto/gossip.proto

```diff
@@ -4,46 +4,133 @@ package kitsune2.gossip;
 
 // A Kitsune2 gossip protocol message.
 message K2GossipMessage {
-  oneof gossip_message {
+  enum GossipMessageType {
```

Comment:
Switched to preferred pattern for enums - it's actually minimal impact on the code and probably easier to evolve so all good there :)